x/simulation/simulate.go

```diff
@@ -232,6 +232,8 @@ func SimulateFromSeed(
 
 		logWriter.AddEntry(EndBlockEntry(blockHeight))
 
+		app.WorkingHash()
```

Comment:
I think we should not make this function public but see if all operations are run in the context of a FinalizeBlock call. Given that this function is called in there